dev/reference/alert-rules.md

```diff
@@ -0,0 +1,1276 @@
+
+#### `TiDB_tikvclient_backoff_count`
+
+* Alert rule:
+
+    `increase(tidb_tikvclient_backoff_count[10m]) > 10`
+
+* Description:
+
+    The number of retries when TiDB fails to access TiKV. When the retry times is over 10 in 10 minutes, an alert is triggered.
+
+* Solution:
+
+    View the monitoring status of TiKV.
+
+#### `TiDB_monitor_time_jump_back_error`
+
+* Alert rule:
+
+    `increase(tidb_monitor_time_jump_back_total[10m]) > 0`
+
+* Description:
+
+    When the time of the machine that holds TiDB rewinds, an alert is triggered.
+
+* Solution:
+
+    Troubleshoot the NTP configurations.
+
+#### `TiDB_ddl_waiting_jobs`
+
+* Alert rule:
+
+    `sum(tidb_ddl_waiting_jobs) > 5`
+
+* Description:
+
+    When the number of DDL tasks pending for execution in TiDB exceeds 5, an alert is triggered.
+
+* Solution:
+
+    Check whether there is any time-consuming `add index` operation that is being executed by running `admin show ddl`.
+
+## PD alert rules
+
+This section gives the alert rules for the PD component.
+
+### Emergency-level alerts
+
+Emergency-level alerts are often caused by a service or node failure. Manual intervention is required immediately.
+
+#### `PD_cluster_offline_tikv_nums`
+
+* Alert rule:
+
+    `sum(pd_cluster_status{type="store_down_count"}) > 0`
+
+* Description:
+
+    PD has not received a TiKV heartbeat for a long time (the default configuration is 30 minutes).
+
+* Solution:
+
+    * Check whether the TiKV process is normal, the network is isolated or the load is too high, and recover the service as much as possible.
+    * If the TiKV cannot be recovered, you can make it offline.
+
+### Critical-level alerts
+
+For the critical-level alerts, a close watch on the abnormal metrics is required.
+
+#### `PD_etcd_write_disk_latency`
+
+* Alert rule:
+
+    `histogram_quantile(0.99, sum(rate(etcd_disk_wal_fsync_duration_seconds_bucket[1m])) by (instance,job,le)) > 1`
+
+* Description:
+
+    etcd writes data to disk with a slower speed than normal. It might lead to PD leader timeout or failure to store TSO on disk in time, which will shut down the service of the entire cluster.
+
+* Solution:
+
+    * Find the cause of slow writes. It might be other services that overload the system. You can check whether PD itself occupies a large amount of CPU or IO resources.
+    * Try to restart PD or manually transfer leader to another PD to recover the service.
+    * Make offline and replace the PD with issues if it cannot be recovered due to environmental factors.
+
+#### `PD_miss_peer_region_count`
+
+* Alert rule:
+
+    `sum(pd_regions_status{type="miss_peer_region_count"}) > 100`
+
+* Description:
+
+    The number of Region replicas is smaller than the value of `max-replicas`. When the TiKV is down and its downtime exceeds `max-down-time`, it usually leads to missing replicas for some Regions during a period of time. When a TiKV node is made offline, it might result in a small number of Regions with missing replicas.
```

Comment:
```suggestion
    The number of Region replicas is smaller than the value of `max-replicas`. When TiKV is down and its downtime exceeds `max-down-time`, it usually leads to missing replicas for some Regions during a period of time. When a TiKV node is made offline, it might result in missing replicas in a small number of Regions.
```